tests/unittest/_torch/speculative/test_torch_rejection_sampling.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+import torch
+from scipy.stats import entropy
+
+from tensorrt_llm._torch.pyexecutor.sampler import (get_rejected_indices,
+                                                    sample_rejected)
+
+
+def test_get_rejected_indices():
```

Comment:
It would be great if we had a way to test the sampler directly instead of testing the algorithm through the helper functions. Unfortunately working with the `LlmRequest` objects outside of `PyExecutor` is a huge pain. This should be good enough for now.